packages/openapi-ts/src/openApi/3.1.x/parser/schema.ts

```diff
@@ -920,7 +920,7 @@ const parseManyTypes = ({
     } else {
       const irTypeSchema = parseOneType({
```

Comment:
`parseOneType` would mutate the `irSchema` passed to it, which causes the object pushed to `schemaItems` all having the same `type`.